test-framework/common/src/main/java/io/quarkus/test/common/DefaultDockerContainerLauncher.java

```diff
@@ -97,6 +99,10 @@ public void start() throws IOException {
         args.add(httpPort + ":" + httpPort);
         args.add("-p");
         args.add(httpsPort + ":" + httpsPort);
+        for (var entry : additionalExposedPorts.entrySet()) {
+            args.add("-p");
+            args.add(entry.getKey() + ":" + entry.getValue());
```

Comment:
Shouldn't the external port be dynamically assigned like for http port ?

Reply:
The configuration allows you to set the host and mapped port. Why would you want to dynamically assign the host port?